### PR TITLE
questionary: app skeleton

### PR DESCRIPTION
**Pull request by VasilMitovClouway**
App sceleton is ready for review.

### CODE REVIEW

File: bank-t2.iml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mgenov) [inline bank-t2.iml]: Project files should not be committed when maven is used.

File: src/main/java/http/servlet/PageHandlerServlet.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    PrintWriter writer = new PrintWriter(new OutputStreamWriter(resp.getOutputStream(), Charsets.UTF_8));
+
+    PageHandler pageHandler = pageRegistry.findHandler(req.getRequestURI());
+
+    PageView pageView = pageHandler.handle(new RequestWrapper(req));
+
+    Template template = configuration.getTemplate(pageView.pageName);
```
Comment (mgenov) [inline src/main/java/http/servlet/PageHandlerServlet.java]: Check the case when template is not found.

File: target/classes/http/servlet/generator_old.html [hunk 3]
```diff
@@ -0,0 +1,96 @@
+<!DOCTYPE html>
```
Comment (mgenov) [inline target/classes/http/servlet/generator_old.html]: one of the generator pages need to be removed as it looks misleading.

File: target/classes/http/servlet/index.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
```
Comment (mgenov) [inline target/classes/http/servlet/index.html]: It looks like more like login page instead of index page.

**Comment by VasilMitovClouway** [comment]
Sceleton changed and ready for review.

File: bank-t2.iml (hunk 1 above)
Comment (mgenov) [inline bank-t2.iml]: .

File: src/test/java/http/servlet/PageHandlerServletTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.HttpURLConnection;
+import java.util.Collections;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class PageHandlerServletTest {
```
Comment (mgenov) [inline src/test/java/http/servlet/PageHandlerServletTest.java]: You need to test and the case when status code is different then HTTP OK (200) and headers and body are passed.

File: src/test/java/http/servlet/RsRedirectTest.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package http.servlet;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class RsRedirectTest {
```
Comment (mgenov) [inline src/test/java/http/servlet/RsRedirectTest.java]: Location header is passed on redirect and status is MOVED_TEMP.

**Comment by mgenov** [comment]
Make sure that commits are squashed before your changes go for the next review.

**Comment by VasilMitovClouway** [comment]
Sceleton is ready, commits are squashed and i will remember to squash them from now on.

File: bank-t2.iml (hunk 1 above)
Comment (mgenov) [inline bank-t2.iml]: .

**Comment by mgenov** [comment]
http://stackoverflow.com/questions/1274057/how-to-make-git-forget-about-a-file-that-was-tracked-but-is-now-in-gitignore

File: src/main/java/core/QuestionRepository.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public interface QuestionRepository {
+  List<Question> getQuestions(Object...objects);
+
+  void register(Question question);
+
+  Question getNew();
```
Comment (mgenov) [inline src/main/java/core/QuestionRepository.java]: Looks un-used.

File: src/main/resources/http/servlet/home.html [hunk 8]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
```
Comment (mgenov) [inline src/main/resources/http/servlet/home.html]: I'm not sure that +'es are required on each row of the template. Seems like wrongly copied from github or somewhere else.

File: src/test/java/http/servlet/RsRedirectTest.java (hunk 6 above)
Comment (mgenov) [inline src/test/java/http/servlet/RsRedirectTest.java]: .

File: target/classes/http/servlet/home.html [hunk 9]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
```
Comment (mgenov) [inline target/classes/http/servlet/home.html]: target foulder should be added to .gitignore

**Pull request merged**